docs/SIL.rst

```diff
@@ -1998,27 +2007,127 @@ This is a form of ownership that is used to model two different use cases:
+and then lower these high level semantics to the operand's ``Ownership
+Constraint`` that we then use to perform OSSA verification.
+
+First with words, we lay out each of the kinds of operand ownership semantics
+grouped according to the types of values that those uses can accept.
+
+Use semantics that require a value with `None`_ ownership:
+
+* **None**: Uses of ownership None. These uses are incompatible with
+  values that have ownership but are otherwise not verified.
+
+Use semantics that require a value with `Guaranteed`_ or `Owned`_
+ownership:
+
+* **Instantaneous Use**: Use an ARC managed value only for the duration of the
+  operation, which may have side effects. (single-instruction `apply`_ with
+  @guaranteed argument)
+
+Use semantics that can accept a value with any form of `Value Ownership Kind`_:
+
+* **Unowned Instantaneous Use**: Use a value without requiring or propagating
+  ownership. The operation may not have side-effects that could affect
+  ownership. This is limited to a small number of operations that are allowed to
+  take Unowned values. (`copy_value`_, single-instruction `apply`_ with @unowned
+  argument))
+
+* **Forwarding Unowned**: Forwarding instruction with an Unowned result. Its
+  operands may have any ownership.
+
+* **Pointer Escape**: Escape a pointer into a value which cannot be tracked or
+  verified.
+
+* **Bitwise Escape**: Escapes the nontrivial contents of the value. OSSA does not
+  enforce the lifetime of the escaping bits.  The programmer must explicitly
+  force lifetime extension.  (`ref_to_unowned`_, `unchecked_trivial_bit_cast`_)
+
+Use semantics that can only accept `Owned`_ values:
+
+* **Borrow**: Propagates the owned value within a scope, without consuming it.
+  (`begin_borrow`_, `begin_apply`_ with @guaranteed argument)
+
+* **Destroying Consume**: Destroys the owned value immediately. (`store`_,
+  `destroy_value`_)
+
+* **Forwarding Consume**: Consumes the owned value indirectly via a move.
+  (`br`_, `destructure_struct`_, `destructure_tuple`_, `tuple`_, `struct`_,
+  `checked_cast_br`_, `switch_enum`_).
+
+Uses that can only accept `Guaranteed`_ values:
+
+* **Nested Borrow**: Propagates the guaranteed value within a nested borrow
+  scope, without ending the outer borrow scope, following stack discipline.
+  (`begin_borrow`_, `begin_apply`_ with @guaranteed).
+
+* **Interior Pointer**: Propagates a trivial value (e.g. address, pointer, or
+  no-escape closure) that depends on the guaranteed value within the base's
+  borrow scope. The verifier checks that all uses of the trivial value are in
+  scope.  (`ref_element_addr`_, `open_existential_box`_)
+
+* **Forwarding Borrow**: Propagates the guaranteed value within the base's
+  borrow scope. (`tuple_extract`_, `struct_extract`_, `checked_cast_br`_,
+  `switch_enum`_)
+
+* **End Borrow**: End the borrow scope opened directly by the operand.  The
+  operand must be a `begin_borrow`_, `begin_apply`_, or function argument.
```

Comment:
I copied/pasted this from what is in tree today. This paragraph is incorrect. I am going to fix it before this goes in.